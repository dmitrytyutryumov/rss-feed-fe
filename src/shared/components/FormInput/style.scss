@import 'src/app/styles/vars.scss';

.form-section {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  gap: .5rem;
  
  &__label {
    text-transform: capitalize;
    font-size: $fs-400;
    line-height: 1;
    font-weight: 500;
  }

  &__input {
    font-family: inherit;
    font-size: $fs-400;
    line-height: 1;

    padding: .6rem .5rem;
    border-radius: .4rem;
    border: $disabled solid 1px;
  }


  &[data-has-error="true"] > &__input, 
  &__input:invalid {
    border: $error solid 2px;
  }
}