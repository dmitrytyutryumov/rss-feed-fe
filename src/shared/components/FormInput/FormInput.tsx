import { useField } from 'formik'

import './style.scss'

export function FormInput({ label, name, className, ...props }) {
  const [field, meta] = useField(name)
  const hasError = meta.touched && meta.error

  return (
    <div
      className={'form-section ' + className || ''}
      data-has-error={Boolean(hasError)}
    >
      {label && (
        <label className="form-section__label" htmlFor="firstName">
          {label}
        </label>
      )}
      <input className="form-section__input" {...field} {...props} />
    </div>
  )
}
