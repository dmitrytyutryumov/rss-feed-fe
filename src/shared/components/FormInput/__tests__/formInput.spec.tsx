import { render } from '@testing-library/react'
import { useField } from 'formik'
import { FormInput } from '../FormInput'

describe('test reset password form', () => {
  beforeEach(() => {
    useField.mockImplementation(() => [{}, { error: 'Error', touched: true }])
  })
  it('test reset password form snapshot', () => {
    // Act
    const { asFragment } = render(
      <form>
        <FormInput name="test" />
      </form>
    )
    // Assert
    expect(asFragment()).toMatchSnapshot()
  })
  it('test reset password form has error', () => {
    // Act
    const container = render(
      <form>
        <FormInput name="test" />
      </form>
    )
    // Assert
    expect(container.container.querySelector('[data-has-error="true"]'))
  })
})
