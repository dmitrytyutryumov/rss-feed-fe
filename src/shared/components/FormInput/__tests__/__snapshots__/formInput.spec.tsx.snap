// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test reset password form test reset password form snapshot 1`] = `
<DocumentFragment>
  <form>
    <div
      class="form-section undefined"
      data-has-error="true"
    >
      <input
        class="form-section__input"
      />
    </div>
  </form>
</DocumentFragment>
`;
